### ISSUE TITLE
[Tiled] Support for Layer Groups

### ISSUE THREAD

**Issue by KuliuTho** [craftworkgames/MonoGame.Extended#435]
Could it be possible to add support for TiledMap layer groups?

**Comment by craftworkgames** [comment]
Sorry, I missed this.

What are layer groups? Are they a new thing in the Tiled editor?

**Comment by KuliuTho** [comment]
They allow for easier sorting of your layers in the Tiled Map Editor. 
**It changes a simple hierarchy like this** 
![Image 1](https://i.gyazo.com/64f02a6c7f11ab91212cfffb39fc70b6.png)
**To this**
![Image 2](https://i.gyazo.com/53a027f670152f58f20e0829f080b757.png)
**It simply just puts a new tag around the existing layers.** 
![Image 3](https://i.gyazo.com/6c98720ce425681045d5d09a436e209e.png)

As far as I can tell they don't seem to be supported in your implementation. (unless I've overlooked something ofc)

**Comment by craftworkgames** [comment]
Cool. Seems useful.

It should be relatively straightforward to add I guess but it doesn't fit into my to-do list right now.  

Would you be interested in making a pull request?

**Comment by craftworkgames** [comment]
@stefanrbk I recall you had this implemented in your fork?

**Comment by ClEngine** [comment]
Yes, I met the problem, too. I really expect the problem to be solved

**Comment by craftworkgames** [comment]
This is definitely on my to-do list (along with a long list of other things). 

Of course, if someone else wants to have a go it's up for grabs.

**Comment by ClEngine** [comment]
Yes, you are right, I will follow you

**Comment by stefanrbk** [comment]
> @stefanrbk I recall you had this implemented in your fork?
I did, but the rendering system isn't compatible any more. I brought it up in issue #481

**Comment by stefanrbk** [comment]
I have implemented crude group layer support in PR #509. It removes the sub-layers from the group and puts them directly in the layer root.
